github/resource_github_tag_protection.go

```diff
@@ -0,0 +1,84 @@
+package github
+
+import (
+	"context"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGithubTagProtection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubTagProtectionCreateOrUpdate,
+		Read:   resourceGithubTagProtectionRead,
+		//Update: resourceGithubTagProtectionCreateOrUpdate,
+		Delete: resourceGithubTagProtectionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"pattern": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tag_protection_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceGithubTagProtectionCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Owner).v3client
+	ctx := context.Background()
+	owner := meta.(*Owner).name
+	repo := d.Get("repository").(string)
+	pattern := d.Get("pattern").(string)
+	log.Printf("[DEBUG] Creating tag protection for %s/%s with pattern %s", owner, repo, pattern)
+	tagProtection, _, err := client.Repositories.CreateTagProtection(ctx, owner, repo, pattern)
+	if err != nil {
+		return err
+	}
+	d.SetId(strconv.FormatInt(tagProtection.GetID(), 10))
+
+	return resourceGithubTagProtectionRead(d, meta)
+}
+
+func resourceGithubTagProtectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Owner).v3client
+	ctx := context.Background()
+	owner := meta.(*Owner).name
+	repo := d.Get("repository").(string)
+	tag_protection, _, err := client.Repositories.ListTagProtection(ctx, owner, repo)
+	if err != nil {
+		return err
+	}
+	d.Set("tag_protection_id", tag_protection[0].GetID())
```

Comment:
what if you have more than one protection on a given "owner/repo" key from listTagProtection? I'm guessing here you'd need to loop thru each and find the one that matches the pattern no?